include/zephyr/net/http/server.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef HTTP2_SERVER_H
+#define HTTP2_SERVER_H
+
+#include <zephyr/posix/poll.h>
+
+#define MAX_CLIENTS 10
+
+struct http2_server_config {
+	int port;
+	int address_family;
+};
+
+struct http2_server_ctx {
+	int sockfd;
+	struct pollfd client_fds[MAX_CLIENTS];
+	int num_clients;
+};
```

Comment:
I would suggest some modification of this (eventually).
```suggestion
struct http2_server_ctx {
	int server_fd; // for accepting new client connections
	int event_fd;
	size_t num_clients;
	struct http2_client_ctx clients[NET_HTTP2_MAX_CLIENTS];
};
```